### DISCUSSION TITLE
"Preview" Button Interface

### DISCUSSION THREAD

**Discussion by alecmev** [directus/directus#3023] [Feature Requests]
I've been unable to find any traces of the [preview functionality](https://docs.directus.io/guides/interfaces.html#preview) in this repo, is there a plan to add it back?

**Comment by rijkvanzanten** [comment]
By "the preview functionality", you mean that interface that renders a link?

**Comment by alecmev** [comment]
Yes, I think? The "preview functionality" above links to the relevant part of the docs for Directus 8.

**Edit:** I mean, an iframe-style integration would be even better, but I'd be happy even with the way it was :slightly_smiling_face:

**Comment by benhaynes** [comment]
I think a configurable button is a good MVP. Adding an option to render that dynamic URL in an iFrame (modal or inline) would be a nice option to add. Also, since many preview sites are auth protected, it would be nice if you could configure/pass basic auth too.

**Reply by alecmev** [reply]
I'm not sure how the auth is normally done in the ecosystem, but at least [Next.js just recommends putting a pre-generated secret in the URL](https://nextjs.org/docs/advanced-features/preview-mode), so a simple template is sufficient.

**Comment by idflood** [comment]
I would love such feature. I don't know if the following details would be covered by this, but the perfect preview feature would have:

- Possibility to preview brand new content which has not been previously saved
- Provide a way to listen for live changes, so that we you change or move something in the backend it is instantly visible in the frontend preview.

**Comment by benhaynes** [comment]
This feature has been implemented, and is an Interface called "Button Links". I will create a new Discussion for an **iFrame Preview Interface** since that would be pretty nice to have!

**Reply by benhaynes** [reply]
Ref: https://github.com/directus/directus/discussions/5897
